### PR TITLE
two minor rhsm followups

### PR DESCRIPTION
**Pull request by cgwalters**
Followups to https://github.com/containers/bootc/pull/977

### CODE REVIEW

File: lib/src/rhsm.rs [hunk 1]
```diff
@@ -96,11 +96,11 @@ pub(crate) async fn publish_facts(root: &Dir) -> Result<()> {
     let (_deployments, host) = crate::status::get_status(&sysroot, booted_deployment.as_ref())?;
 
     let facts = RhsmFacts::from(host.status);
-    let mut bootc_facts_file = root.open_with(
-        FACTS_PATH,
-        OpenOptions::new().write(true).create(true).truncate(true),
-    )?;
-    serde_json::to_writer_pretty(&mut bootc_facts_file, &facts)?;
+    root.atomic_replace_with(FACTS_PATH, |w| {
```
Comment (jeckersb) [inline lib/src/rhsm.rs:99]: :+1: TIL!

**Review by jeckersb — APPROVED** [review]

**Comment by cgwalters** [comment]
Ug, the [integration test](https://artifacts.dev.testing-farm.io/db58b8df-3f29-4dd8-a686-d3db5c65d256/) on this run hit the dreaded 
> ERROR Switching: Pulling: Importing: Parsing layer blob sha256:8a3d9214708cd83d1dcc9b40fb6363f838bad5e57980cbbcfc5aa50e5b487beb: failed to invoke method FinishPipe: failed to invoke method FinishPipe: write |1: broken pipe

too...

**Comment by jeckersb** [comment]
> Ug, the [integration test](https://artifacts.dev.testing-farm.io/db58b8df-3f29-4dd8-a686-d3db5c65d256/) on this run hit the dreaded
> 
> > ERROR Switching: Pulling: Importing: Parsing layer blob sha256:8a3d9214708cd83d1dcc9b40fb6363f838bad5e57980cbbcfc5aa50e5b487beb: failed to invoke method FinishPipe: failed to invoke method FinishPipe: write |1: broken pipe
> 
> too...

This is the equivalent of getting a stocking full of coal for Christmas.  I am not looking forward to fighting this again in 2025!

**Comment by cgwalters** [comment]
On this topic, I have now marked the integration tests as required.

**Pull request merged**